app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java

```diff
@@ -49,6 +49,7 @@
                 this.serviceLocator.postOnEventBus(new NewDataReceivedEvent(object.getData(), object.getClazz()));
             } catch (final ClassNotFoundException | IOException e) {
                 LOGGER.error(" Couldn't start deserialization!", e);
+                throw new DeserializationException(e);
```

Comment:
The only place where objects are deserialized is after they are received from another device over network. All incoming network traffic is handled by DataReceiver from APairingHandler